app/lib/features/chat_ng/widgets/events/chat_event.dart

```diff
@@ -0,0 +1,199 @@
+    final avatarInfo = ref.watch(
+      memberAvatarInfoProvider(
+        (
+          roomId: roomId,
+          userId: item.sender(),
+        ),
+      ),
+    );
+    final options = AvatarOptions.DM(avatarInfo, size: 14);
+    final myId = ref.watch(myUserIdStrProvider);
+    final messageId = msg.uniqueId();
+    final isUser = myId == item.sender();
+    // TODO: render a regular timeline event
+    return Row(
+      mainAxisAlignment:
+          !isUser ? MainAxisAlignment.start : MainAxisAlignment.end,
+      crossAxisAlignment: CrossAxisAlignment.end,
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        (showAvatar && !isUser)
+            ? Padding(
+                padding: const EdgeInsets.only(left: 8),
+                child: ActerAvatar(options: options),
+              )
+            : const SizedBox(width: 40),
+        Flexible(
+          child: _buildEventItem(messageId, item, isUser, showAvatar),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildEventItem(
+    String messageId,
+    RoomEventItem item,
+    bool isUser,
+    bool showAvatar,
+  ) {
+    final eventType = item.eventType();
+    switch (eventType) {
+      case 'm.room.message':
+        return _buildMsgEventItem(roomId, messageId, item, isUser, showAvatar);
+      case 'm.room.redaction':
+        return ChatBubble(
+          isUser: isUser,
+          showAvatar: showAvatar,
+          child: RedactedMessageWidget(),
+        );
+      case 'm.room.encrypted':
+        return ChatBubble(
+          isUser: isUser,
+          showAvatar: showAvatar,
+          child: EncryptedMessageWidget(),
+        );
+      case 'm.room.member':
+        return MemberUpdateEvent(isUser: isUser, item: item);
+      case 'm.policy.rule.room':
+      case 'm.policy.rule.server':
+      case 'm.policy.rule.user':
+      case 'm.room.aliases':
+      case 'm.room.avatar':
+      case 'm.room.canonical_alias':
+      case 'm.room.create':
+      case 'm.room.encryption':
+      case 'm.room.guest_access':
+      case 'm.room.history_visibility':
+      case 'm.room.join_rules':
+      case 'm.room.name':
+      case 'm.room.pinned_events':
+      case 'm.room.power_levels':
+      case 'm.room.server_acl':
+      case 'm.room.third_party_invite':
+      case 'm.room.tombstone':
+      case 'm.room.topic':
+      case 'm.space.child':
+      case 'm.space.parent':
+        return StateUpdateEvent(item: item);
+      default:
+        return _buildUnsupportedMessage(eventType);
+    }
+  }
+
+  Widget _buildMsgEventItem(
+    String roomId,
+    String messageId,
+    RoomEventItem item,
+    bool isUser,
+    bool showAvatar,
+  ) {
+    final msgType = item.msgType();
+    final content = item.msgContent();
+
+    // shouldn't happen but in case return empty
+    if (msgType == null || content == null) return const SizedBox.shrink();
+
+    switch (msgType) {
+      case 'm.emote':
```

Comment:
shouldn't we make is a separate widget as well now? I'd argue that an emote and sticker shouldn't have the bubble for example ...